eventcrypto.go

```diff
@@ -184,3 +184,55 @@ func verifyEventSignature(signingName string, keyID KeyID, publicKey ed25519.Pub
 
 	return VerifyJSON(signingName, keyID, publicKey, redactedJSON)
 }
+
+// VerifyEventSignatures checks that each event in a list of events has valid
+// signatures from the server that sent it.
+func VerifyEventSignatures(events []Event, keyRing KeyRing) error {
+	var toVerify []VerifyJSONRequest
+	for _, event := range events {
+		redactedJSON, err := redactEvent(event.eventJSON)
+		if err != nil {
+			return err
+		}
+		v := VerifyJSONRequest{
+			Message:    redactedJSON,
```

Comment:
Naming and docs: It's not clear that `VerifyJSONRequest` actually wants the redacted JSON and not the full event JSON. I would kill two birds with one stone and `s/Message/RedactedJSON/` assuming `VerifyJSONRequest` is only ever used for `Event` structs.